Generate xml

jmgrady opened the pull request:
Implement generation of XML for from the `.docx` files

jmgrady committed:
Initial XML generation framework

jmgrady committed:
Add printing and updating methods to semantic_domain_subs.py

jmgrady committed:
Add copy and update methods for semantic_domain_subs

jmgrady committed:
Implement removal of old_lang

jmgrady committed:
Fix help text

imnasnainaec commented on py-doc-to-xml/docx_to_xml/semantic_domain_xml.py:
```diff
@@ -0,0 +1,92 @@
+                )
+        else:
+            # The source document does not have the current domain so
+            # we copy the English text.
```
Let's throw a warning in this case.

imnasnainaec replied:
And let's only copy the Abbreviation from en... the other fields that still need to be translated should be left empty `ml` or `ta` elements.

jmgrady replied:
Warning added.

Are you sure about leaving the fields blank?  In _The Combine_ UI any untranslated strings get the English version.  In crowd-in, will the English be visible?  What will the app do if there is a blank domain definition?  If we have the English, then at least the app is functional before translation is complete and the translated domain text can be viewed in _The Combine_.

jmgrady replied:
If blank fields are what we want, do we want a single blank question if there are multiple questions or a blank question for each existing question in English?  Also, any of the options is straight-forward to implement and is fine with me.  I just want to confirm the choice.

imnasnainaec replied:
I believe we want a blank for every question.

imnasnainaec replied:
The the process of generating the JSON file from Crowdin uses the English when a translation is mission. For uploading to Crowdin, I think we want a blank field in the XML for everything that still needs to be translated.

jmgrady replied:
Done.

imnasnainaec commented on py-doc-to-xml/docx_to_xml/semantic_domain_xml.py:
```diff
@@ -0,0 +1,92 @@
+            new_len = len(domain.questions)
+            if node_len != new_len:
+                print(
+                    f"WARNING: Number of questions for {domain_abbr}, 'es' - {node_len}; {lang} - {new_len}",
```
Shouldn't this 'es' be 'en'?

jmgrady replied:
Yes.  Fixed.

jmgrady committed:
Restore --debug-file option

jmgrady committed:
Update regex to use ascii 0-9 instead of digit escape sequence.

The patterns used to detect a semantic domain number or a number
enumerating a question were updated to replace '\d' with '[0-9]' to
limit matches to the Roman 0-9 characters.

jmgrady committed:
Fix storing of example words and example sentences in parser.

jmgrady committed:
Fix parsing of example words & sentences

jmgrady committed:
Fix Python formatting and type hints

jmgrady committed:
Make message a one-liner for too many text blocks in a question

jmgrady committed:
Fix abbreviation for English in warning message

jmgrady committed:
Merge branch 'main' into generate_xml

# Conflicts:
#	py-doc-to-xml/docx_to_xml/doc1.py
#	py-doc-to-xml/docx_to_xml/util.py

jmgrady committed:
Fix typos in comments

jmgrady committed:
Add blank fields for domains missing from source document

jmgrady committed:
Fix comment

jmgrady committed:
Fix Python formatting & import statements

imnasnainaec commented on py-doc-to-xml/docx_to_xml/util.py:
```diff
@@ -42,7 +42,7 @@ def split_question(s: str) -> Tuple[str, str]:
      (1)
     """
-    return split_line(s, pattern=r"\s*\(?(\d+)[).]?")
+    return split_line(s, pattern=r"\s*\(?([0-9]+)[).]?")
```
Can we change the `[).]?` to `[). ]` so it doesn't catch (for example) 8.4.4 question 4 example word "5:00".

jmgrady replied:
I thought that there were cases where there was no whitespace between the question number and the question.  Even so, the question number should end in a '.' or ')' so I think we just want to get rid of the second '?' in the pattern.

imnasnainaec replied:
Yes, that should be fine.

jmgrady committed:
Make . or ) required at the end of a question
